dev/provider_packages/refactor_provider_packages.py

```diff
@@ -146,6 +146,13 @@ def rename_deprecated_modules(self) -> None:
         for new, old in changes:
             self.qry.select_module(new).rename(old)
 
+        def is_not_k8spodop(node: LN, capture: Capture, filename: Filename) -> bool:
+            return not filename.endswith("/kubernetes_pod.py")
+
+        self.qry.select_module("airflow.providers.cncf.kubernetes.backcompat").filter(
+            callback=is_not_k8spodop
+        ).rename("airflow.kubernetes")
+
```

Comment:
```suggestion
        self.qry.select_module("airflow.providers.cncf.kubernetes.backcompat.pod_runtime_info")\
              .rename("airflow.kubernetes.pod_runtime_info")
```

That + restoring class removal should fix it.

Reply:
Does select_module just operate on that one module, not sub modules under it?

Reply:
There are none under it :). It's the full package name not the module. But likely it should be select_package indeed :)

Reply:
@potiuk @ashb wait can someone please break this down for me? I'm still confused as to what is broken.

Reply:
@dimberman: Not broken, but can be improved :) 

This commit: https://github.com/apache/airflow/commit/8d0be15ae4e4c1cd51d61ccd71fe0a8bef7f0ffe removes the backcompat/pod.py and other files from backports as they are not needed there. Except that 

`from airflow.kubernetes.pod_runtime_info_env import PodRuntimeInfoEnv` in `kubernetes_pod.py` still points to backcompat because this file is excluded from the rename above.

The "perfect" fix is to:
1) Also replace the import in `./kubernetes_pod.py`. Can be likely done with this without affecting the other imports in ./kubernetes.pod:

```
        self.qry.select_package("airflow.providers.cncf.kubernetes.backcompat.pod_runtime_info")\
              .rename("airflow.kubernetes.pod_runtime_info")
```

2) remove the backcompat files as in https://github.com/apache/airflow/commit/8d0be15ae4e4c1cd51d61ccd71fe0a8bef7f0ffe

Reply:
ohhh ok. I think I get it. 

So basically the idea here is that we want to delete the backcompat files in 1.10 since we're not using them, though we were running into a bowler issue because we weren't changing one import in `kubernetes_pod.py` (this was because we needed to keep the original import for the converters.  So adding that extra query fixes the `pod_runtime_info_env` import and allows us to delete those files.